src/pool-service/IrcClientRedisState.ts

```diff
@@ -21,20 +36,50 @@ interface IrcClientStateDehydrated {
     capabilities: ReturnType<IrcCapabilities["serialise"]>;
     supportedState: IrcSupported;
     hostMask: string;
-    chans: [string, ChanData][];
+    chans: [string, ChanDataDehydrated][];
     prefixForMode: {
         [mode: string]: string;
     };
     maxLineLength: number;
     lastSendTime: number;
 }
 
+
 export class IrcClientRedisState implements IrcClientState {
-    private putStatePromise: Promise<void> = Promise.resolve();
+    private putStatePromise: Promise<unknown> = Promise.resolve();
 
-    static async create(redis: Redis, clientId: string) {
-        const data = await redis.hget(REDIS_IRC_CLIENT_STATE_KEY, clientId);
+    static async create(redis: Redis, clientId: string, freshState: boolean) {
+        log.debug(`Requesting ${freshState ? "fresh" : "existing"} state for ${clientId}`);
+        const data = freshState ? null : await redis.hget(REDIS_IRC_CLIENT_STATE_KEY, clientId);
         const deseralisedData = data ? JSON.parse(data) as IrcClientStateDehydrated : {} as Record<string, never>;
+        const chans = new Map<string, ChanData>();
+
+        // In a previous iteration we failed to seralise this properly.
+        const isPreviousBuggyState = !Array.isArray(deseralisedData.chans?.[0]?.[1]?.users);
+        deseralisedData?.chans?.forEach(([channelName, chanData]) => {
```

Comment:
TIL `JSON.parse` can return `null`, but only when it's given `null` or a string of `"null"`.  So for consistency with the rest of nearby code & the type hints on `deserialisedData`, let's not bother with optional chaining on it:
```suggestion
        deseralisedData.chans?.forEach(([channelName, chanData]) => {
```